src/App.vue

```diff
@@ -1,20 +1,24 @@
 <template>
   <div id="app">
+    <dynamic-component name="Header" />
     <h1>Multibrand app / {{brand}}</h1>
+    <dynamic-component name="Box" />
     <branded-button type="primary">Hi there! This is a branded button</branded-button>
   </div>
 </template>
 
 <script>
-import BrandedButton from './components/BrandedButton.vue'
+import BrandedButton from '@/components/BrandedButton'
+import DynamicComponent from '@/components/DynamicComponent'
```

Comment:
Note that I've structured the folders/files in a way that these imports look exactly the same, however `BrandedButton` will resolve to `src/components/BrandedButton.vue` while as 
`DynamicComponent` will resolve to `src/components/DynamicComponent/index.js`